File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/annotation/Password.java
package com.windea.demo.cloudcollect.core.validation.annotation;

import com.windea.demo.cloudcollect.core.validation.validator.PasswordValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * 密码的校验注解。
 */
@Documented
@Constraint(validatedBy = {PasswordValidator.class})
@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
@Retention(RUNTIME)
public @interface Password {
	String message() default "{javax.validation.constraints.Pattern.message}";

	Class<?>[] groups() default {};

	Class<? extends Payload>[] payload() default {};
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/annotation/Password.kt
package com.windea.demo.cloudcollect.core.validation.annotation

import com.windea.demo.cloudcollect.core.validation.validator.*
import javax.validation.*
import kotlin.annotation.AnnotationTarget.*
import kotlin.reflect.*

/**密码的校验注解。*/
@MustBeDocumented
@Constraint(validatedBy = [PasswordValidator::class])
@Target(FIELD, AnnotationTarget.PROPERTY_GETTER)
annotation class Password(
	val message: String = "Password is not correct.",
	val groups: Array<KClass<*>> = [],
	val payload: Array<KClass<out Payload>> = []
)
